tests/devices/test_default_qubit.py

```diff
@@ -2529,68 +2546,33 @@ def test_error_hamiltonian_expval_finite_shots(self):
-        def circuit(x, y):
-            qml.RX(x, wires=0)
-            qml.RX(y, wires=1)
-            return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))]
-
-        out = circuit(x, y)
-        expected = qml.math.stack([qml.math.cos(x) * qml.math.ones_like(y), -qml.math.sin(y)]).T
-
-        assert circuit.device.num_executions == len(y)
-        tol = 1e-10 if shots is None else 1e-2
-        assert qml.math.allclose(out, expected, atol=tol, rtol=0)
-
-    @pytest.mark.parametrize("x, y", [(0.2, [0.4]), ([0.1, 5.1], [0.1, -0.3])])
-    @pytest.mark.parametrize("shots", [None, 1000000])
-    def test_with_Hamiltonian(self, x, y, shots):
-        """Test that broadcasting on a circuit with a
-        single parametrized operation works."""
-        dev = qml.device("default.qubit", wires=2, shots=shots)
-
-        H = qml.Hamiltonian([0.3, 0.9], [qml.PauliZ(0), qml.PauliY(1)])
-        H.compute_grouping()
+class TestGetBatchSize:
+    """Tests for the helper method ``_get_batch_size`` of ``QubitDevice``."""
 
-        @qml.qnode(dev)
-        def circuit(x, y):
-            qml.RX(x, wires=0)
-            qml.RX(y, wires=1)
-            return qml.expval(H)
-
-        out = circuit(x, y)
-        expected = 0.3 * qml.math.cos(x) * qml.math.ones_like(y) - 0.9 * qml.math.sin(y)
-
-        assert circuit.device.num_executions == len(y)
-        tol = 1e-10 if shots is None else 1e-2
-        assert qml.math.allclose(out, expected, atol=tol, rtol=0)
+    @pytest.mark.parametrize("shape", [(4, 4), (1, 8), (4,)])
+    def test_batch_size_None(self, shape):
+        """Test that a ``batch_size=None`` is reported correctly."""
+        dev = qml.device("default.qubit", wires=1)
+        tensor0 = np.ones(shape, dtype=complex)
+        assert dev._get_batch_size(tensor0, shape, qml.math.prod(shape)) is None
+        tensor1 = np.arange(np.prod(shape)).reshape(shape)
+        assert dev._get_batch_size(tensor1, shape, qml.math.prod(shape)) is None
+
+    @pytest.mark.parametrize("shape", [(4, 4), (1, 8), (4,)])
+    @pytest.mark.parametrize("batch_size", [1, 3])
+    def test_batch_size_int(self, shape, batch_size):
+        """Test that an integral ``batch_size`` is reported correctly."""
+        dev = qml.device("default.qubit", wires=1)
+        full_shape = (batch_size,) + shape
+        tensor0 = np.ones(full_shape, dtype=complex)
+        assert dev._get_batch_size(tensor0, shape, qml.math.prod(shape)) == batch_size
+        tensor1 = np.arange(np.prod(full_shape)).reshape(full_shape)
+        assert dev._get_batch_size(tensor1, shape, qml.math.prod(shape)) == batch_size
+
+    @pytest.mark.filterwarnings("ignore:Creating an ndarray from ragged nested")
+    def test_invalid_tensor(self):
+        """Test that an error is raised if a tensor is provided that does not
+        have a proper shape/ndim."""
+        dev = qml.device("default.qubit", wires=1)
+        with pytest.raises(ValueError, match="could not broadcast"):
```

Comment:
Ah! This is a `numpy` default error. This test is a bit tricky:
1. With tensorflow, `_get_batch_size` catches two types of errors (because it needs to in order to support `tf.function`)
2. One of the two types that are caught is `ValueError`. 
3. However, we do *not* want to catch `ValueError` if we are not using `tf.function`. This "not catching of `ValueError`" is tested here :) 
The logic is in the `except` clause in `_get_batch_size`.